tests/unit/test_charm.py

```diff
@@ -177,9 +233,120 @@ def test_provider(self, _create_ranger_service):
 
         relation_data = self.harness.get_relation_data(rel_id, "ranger-k8s")
         assert relation_data == {
-            "policy_manager_url": "http://ranger-k8s:6080"
+            "policy_manager_url": "http://ranger-k8s:6080",
+            "service_name": "trino-service",
         }
 
+    @mock.patch("charm.RangerProvider._create_ranger_service")
+    def test_user_group_configuration(self, _create_ranger_service):
+        """The user-group-configuration paramerter is validated."""
+        harness = self.harness
+        simulate_lifecycle(harness)
+        rel_id = harness.add_relation("policy", "trino-k8s")
+        harness.add_relation_unit(rel_id, "trino-k8s/0")
+
+        event = make_policy_relation_changed_event(rel_id)
+        harness.charm.provider._on_relation_changed(event)
+
+        # Unable to parse configuration file.
+        self.harness.update_config(
+            {"user-group-configuration": f"{INCORRECTLY_FORMATTED_CONFIG}"}
```

Comment:
Are the constants not already strings? Why not just `{"user-group-configuration": INCORRECTLY_FORMATTED_CONFIG}`?

Reply:
You are correct this is not needed, I have removed the f strings in all these types of tests.